src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/MobEventsUtils.java

```diff
@@ -116,20 +180,24 @@ else if (hordeTotal > MAX_SIZE)
      */
     private static BlockPos calculateSpawnLocation(final World world, final Colony colony)
     {
-        final ColonyView colonyView = ColonyManager.getClosestColonyView(world, colony.getCenter());
-        if (colonyView == null)
+        if (colony == null)
         {
+            Log.getLogger().info("When trying to find SpawnPoint for Barbarian Event, the Colony was Null!");
+            Log.getLogger().info("LIKE, PLEASE REPORT THIS, THIS MAKES NO FREAKING SENSE!");
             return null;
         }
-        final BlockPos center = colonyView.getCenter();
-        final int radius = Configurations.workingRangeTownHall;
 
+        final BlockPos center = colony.getCenter();
+        final int radius = Configurations.workingRangeTownHall;
         final int randomDegree = world.rand.nextInt((int) WHOLE_CIRCLE);
 
         final double rads = (double) randomDegree / HALF_A_CIRCLE * Math.PI;
+
         final double x = Math.round(center.getX() + radius * Math.sin(rads));
         final double z = Math.round(center.getZ() + radius * Math.cos(rads));
-        return world.getTopSolidOrLiquidBlock(new BlockPos(x, center.getY(), z));
+
+        final BlockPos topBlock = BlockPosUtil.findLand(new BlockPos(x, center.getY(), z), world);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "topBlock". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1488)